### PR TITLE
vite-plugin: Add vite 3 to peer dependencies

### LINKED ISSUE #760: Vite Plugin Peer Dependency Prevents Vite 3.0
### Describe the bug

Now that [Vite 3 is out](https://vitejs.dev/blog/announcing-vite3.html), there don't appear to be any changes preventing Vanilla Extract from supporting `"vite": "^2.2.3 || ^3.0.0-0"` as a peer dependency.

Without updating the peer dependency in this repo, and attempting to install the Vite 3 gives:
```
npm ERR! Found: vite@3.0.2
npm ERR! node_modules/vite
npm ERR!   dev vite@"^3.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer vite@"^2.2.3" from @vanilla-extract/vite-plugin@3.2.1
npm ERR! node_modules/@vanilla-extract/vite-plugin
npm ERR!   dev @vanilla-extract/vite-plugin@"^3.2.1" from the root project
```

### Reproduction

Reproduction

### System Info

```shell
System:
    OS: macOS 12.2.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 2.33 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.asdf/installs/nodejs/16.13.1/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.5.5 - ~/.asdf/plugins/nodejs/shims/npm
```

### Used Package Manager

npm

### Logs

```shell
npm ERR! Found: vite@3.0.2
npm ERR! node_modules/vite
npm ERR!   dev vite@"^3.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer vite@"^2.2.3" from @vanilla-extract/vite-plugin@3.2.1
npm ERR! node_modules/@vanilla-extract/vite-plugin
npm ERR!   dev @vanilla-extract/vite-plugin@"^3.2.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/seek-oss/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by tom-sherman**
Closes #760

### CODE REVIEW

**Comment by mattcompiles** [comment]
@tom-sherman Thanks for the PR. We had a crack at updating to Vite 3 the other day but some issues were caused by the upgrade due to ESM stuff I think. I'm happy to bump the peer dep, but the dev dep version will have to stay as v2 for the time being.

**Comment by tom-sherman** [comment]
@mattcompiles cool, I'll remove the dev dep change.

Could we discuss the errors in another issue then? Happy to have a go at solving them.

File: .changeset/stale-feet-sniff.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@vanilla-extract/vite-plugin': minor
```
Comment (mattcompiles) [inline .changeset/stale-feet-sniff.md:2]:
```suggestion
'@vanilla-extract/vite-plugin': patch
```

File: .changeset/stale-feet-sniff.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Adds vite v3 to peer dependencies. Fixes https://github.com/seek-oss/vanilla-extract/issues/760
```
Comment (mattcompiles) [inline .changeset/stale-feet-sniff.md:5]: Don't think the issue reference is needed here. 

```suggestion
Adds vite v3 to peer dependencies
```

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -574,7 +574,7 @@ packages:
       gensync: 1.0.0-beta.2
```
Comment (mattcompiles) [inline pnpm-lock.yaml:574]: Not sure why there is a lock file change here? Could we remove it?

**Review by mattcompiles — COMMENTED** [review]
Thanks bud. Just a couple of small comments.

**Review by mattcompiles — APPROVED** [review]

**Pull request merged**